proposals/dark_mode_extension_icons.md

```diff
@@ -0,0 +1,205 @@
+  tabId?: number,
+  variants?: {
+    "<dark|light>": <numberToPathDictionary>
+  },
+});
+```
+
+### Examples
+
+manifest.json
+```
+{
+  "icon_variants": {
+    "dark": {
+      "128": "128_dark.png"
+    },
+    "light": {
+      "128": "128_light.png"
+    }
+  },
+  "action": {
+    "default_icon": {
+      "128": "128_action.png"
+    },
+    "icon_variants": {
+      "dark": {
+        "128": "128_action_dark.png"
+      },
+      "light": {
+        "128": "128_action_light.png"
+      }
+    }
+  }
+}
+```
+
+Action
+```
+action.setIcon({
+  variants?: {
+    "dark": {
+      "128": "128_action_dark.png"
+    },
+    "light": {
+      "128": "128_action_light.png"
+    }
+  },
+});
+```
+
+### Behavior
+
+Existing manifest key. The behavior of the icons manifest key will remain unchanged.
+```
+{
+  "icons": {
+    "64": "icon_64.png"
+  },
+  "action": {
+    "default_icon": {
+      "64": "action_64.png"
+    }
+  }
+}
+```
+
+Order of precedence. The new `icon_variants` keys and subkeys will take
+precedent, followed by the incumbent `icons` key.
+
+### New Permissions
+
+N/A.
+
+### Manifest File Changes
+
+#### Optional or required?
+`icon_variants` is an optional key of the top level manifest dictionary.
+`icon_variants` is also an optional sub-key in the action key dictionary.
+
+#### Fallback
+`icon_variants` will be used if they're supplied. Otherwise `default_icon` will
+be used if it's supplied. If neither are supplied, then `icons` will be used.
+
+#### Warning or error?
+The new `icon_variants` key (or sub-key) will only emit a non-blocking warning
+rather than a hard error in the event of unexpected values or missing icons.
+This will allow this optional key to be more flexible in the event of ideas for
+future changes. The `icons` key cannot be used for dark mode support because it
+errors in cases when a warning would have otherwise sufficed. The warning is
+ok for `icon_variants` based on the safety net outlined in the fallback section.
+
+## Security and Privacy
+
+### Exposed Sensitive Data
+
+N/A.
+
+### Abuse Mitigations
+
+N/A.
+
+### Additional Security Considerations
+
+N/A.
+
+## Alternatives
+
+### Existing Workarounds
+
+1. A developer could ask the user to specify what mode they're in and then
+dynamically set the icon to a dark mode icon using action.setIcon(). That
+wouldn't change the management page icon.
+2. A developer could change icons dynamically to dark mode icons if this is
+true: `window.matchMedia('(prefers-color-scheme: dark)')`.
+
+### Open Web API
+
+As stated in the workarounds section, the following is already an option to consider: `window.matchMedia('(prefers-color-scheme: dark)')`. However, that
+wouldn't automatically set icons dynamically, as it would require subsequent
+calls to action.setIcon(). It also wouldn't update the icon on the management
+pages.
+
+## Implementation Notes
+
+N/A.
+
+## Future Work
+
+Svg support may be considered at some point for some browsers.
```

Comment:
I would probably drop this, it doesn't feel related to the proposal (as much as I would love it!).

Reply:
Done.